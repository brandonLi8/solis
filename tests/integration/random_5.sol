# Copyright Â© 2022-2023 Brandon Li. All rights reserved.

##
Integration test of a randomly generated program.
NOTE: this file was auto-generated with the `random_program_gen.py` script.
##

if !(false) {
  -(16.705120138091935);
  let a1: float = 39.110443040838355;
  14.966534400017414;
  let b1: float = a1;
  !((!(true)) == (false))
} else if (18.77538286404134) == (-(33.63055955688019)) {
  let c1: float = 48.856050206020534;
  262;
  let d1: int = 609;
  let e1: float = 27.97950150182125;
  let f1: bool = true;
  let g1: bool = !(false);
  false
} else if if !(true) {
  let h1: bool = !((0.8158990496445864) > (-(989)));
  -(485);
  true
} else if !((35.894599314660304) < (-(70))) {
  -(27.902689174041843);
  -(15.584889541537796);
  if (27.417983918507552) < (-(44.989047436064176)) {
    !(false);
    true;
    !(true)
  } else if (14.62389036929463) == (-(6.785659593548867)) {
    !(false);
    let i1: float = -(6.147191496122966);
    !(true);
    789;
    let j1: int = -(561);
    false
  } else if false {
    !(false)
  } else if true {
    let k1: bool = !(false);
    -((7.976925015010021) / (-(49.47371226461031)));
    (-(912)) > (-(24.05308938030435));
    let l1: int = -(352);
    !(k1);
    k1
  } else {
    let m1: bool = (40.507167200283206) == (-((-(49.58253774433996)) - ((-(15.504324950702886)) - (-(37.51291945412526)))));
    -(26.370910512937026);
    486;
    !(true);
    let n1: float = -(25.37769845125685);
    -(777);
    !(m1)
  }
} else if !(true) {
  -(92);
  (16.416566550384296) != (12.986585026070863)
} else if !((95) < (779)) {
  -(17.58874297383296);
  if (-(728)) >= (-(111)) {
    -(953)
  } else if !((885) == (8)) {
    -(32.71591915246596);
    let o1: int = 392;
    let p1: bool = true;
    o1;
    let q1: int = o1;
    let r1: bool = !(true);
    o1
  } else if false {
    let s1: float = 29.4755847733668;
    let t1: bool = true;
    (808) - (215)
  } else if !((-(789)) >= (-(864))) {
    let u1: float = (226) / (21.847962819133087);
    let v1: int = 821;
    -(v1);
    v1
  } else {
    if !(false) {
      46.021775630057284;
      let w1: int = -(713);
      let x1: int = w1;
      let a2: bool = true;
      254;
      let b2: int = -(676);
      let c2: bool = !(false);
      w1 + b2 + x1
    } else if false {
      let d2: bool = true;
      -(45.73884412449855);
      38.87557247933562;
      d2;
      (594) * (-(610))
    } else if !(true) {
      -(145);
      -(45.76741434610104);
      let e2: int = -(404);
      !((782) == (507));
      e2
    } else if true {
      (528) / (-(502));
      (false) == (!(true));
      let f2: int = -(33);
      let g2: int = -((-((f2) - ((-(f2)) - (-(529))))) % (f2));
      g2
    } else {
      let h2: bool = !(true);
      let i2: int = -(76);
      i2
    }
  };
  !((185) < (-(((-(989)) + ((-(47.82689314075171)) - (754))) - (((21.975975803278576) / (-(927))) * (-(31.190403214221313))))));
  !(false)
} else {
  !(true);
  true
} {
  -(869);
  !(false)
} else if true {
  (87) == (549)
} else {
  -((-(896)) * (142));
  let j2: float = -(22.501847116636057);
  let k2: float = 30.00197298448046;
  -(96);
  -(k2);
  let l2: int = 869;
  !(false);
  (!((46.51461246811852) >= (-(14.960041736868765)))) != (true)
};
let m2: bool = !(false);
if !(if !(true) {
  true
} else {
  let n2: int = 62;
  let o2: int = -(n2);
  let p2: float = -(27.965655820184985);
  !(m2);
  -(p2);
  -(if m2 {
    !((-(7.142102799117537)) > (65));
    -(45.54716323973687);
    let q2: float = -((-(n2)) + (26.468049436865645));
    let r2: bool = !(true);
    false;
    let s2: float = p2;
    n2
  } else if true {
    -(901);
    let t2: float = -(0.06625835456048562);
    let u2: int = -(788);
    !(m2);
    let v2: int = -(213);
    41.45820784528552;
    -(30.436409962596574);
    !(m2);
    n2 + o2
  } else if (312) >= (o2) {
    m2;
    829;
    20.038468375380273;
    -(p2);
    o2
  } else if !((o2) != (564)) {
    m2;
    let w2: int = n2;
    w2;
    let x2: bool = m2;
    let a3: float = 20.73902225825819;
    w2 + n2 + o2
  } else {
    -(2.4540219264254013);
    let b3: float = (34.22231180757434) - (-(n2));
    !(m2);
    let c3: bool = (!(true)) != (m2);
    807;
    n2
  });
  !(m2)
}) {
  let d3: bool = m2;
  512;
  18.02766536732312;
  (74) % (-(41))
} else if !(m2) {
  !((92) != (-(86)));
  let e3: bool = m2;
  let f3: float = -(41.72192774736748);
  -(3.492842950323351);
  false;
  841
} else {
  520
};
-(166);
15.156197234430989;
-(if !(m2) {
  let g3: bool = !(m2);
  28.499606367339766;
  47.98690277042299;
  let h3: float = 40.02566106284743;
  let i3: bool = !(false);
  -(301)
} else if true {
  504
} else {
  45.56135545834342;
  let j3: int = -(91);
  let k3: float = 12.16135563119165;
  true;
  j3
});
true;
let l3: float = 4.768716268888914;
(-(46.546329051891746)) - (((559) + (-(143))) + (if !(m2) {
  let m3: int = 15;
  let n3: float = -(11.760521405552277);
  m3
} else {
  let o3: bool = true;
  m2;
  -(279)
}));
let p3: bool = true;
-((682) + (-(554)));
-((if !(false) {
  20.3736614353474;
  let q3: int = 951;
  !(false);
  99;
  -(l3);
  -(98);
  -(l3);
  -(436);
  q3
} else if !(if m2 {
  16.016319208029483;
  -((426) - (-(853)));
  -(354);
  -(l3);
  718;
  !(true)
} else if !((959) <= (-(295))) {
  let r3: float = l3;
  33.10103806589315;
  let s3: float = -(17.57500931478712);
  -(40);
  !(p3)
} else if true {
  !(m2);
  p3;
  let t3: float = l3;
  -(73);
  -((-(905)) + (276));
  -(137);
  let u3: bool = true;
  true
} else {
  let v3: bool = !((-(48.684529348030416)) <= ((-(l3)) / (160)));
  l3;
  !((-(27.24384588812835)) < (8.782379366251853))
}) {
  -((851) - (-(617)));
  -(742);
  -(149);
  let w3: bool = !(false);
  let x3: int = -(946);
  let a4: int = -((-((x3) % (x3))) - (x3));
  !(false);
  (10.39033812062014) <= (-(l3));
  x3
} else {
  let b4: float = 5.5651375886060945;
  !(p3);
  -((-(b4)) / ((891) * (-(401))));
  !(p3);
  -((-(392)) * ((-(569)) * (-(if !(false) {
    -(300);
    let c4: bool = (4.766885639994846) >= (-(b4));
    (-(b4)) > (l3);
    -(724);
    44.35524484036894;
    528;
    m2;
    p3;
    -(742)
  } else {
    let d4: bool = !(false);
    951;
    let e4: bool = !(true);
    let f4: int = -((-(936)) / (53));
    f4
  }))))
}) - (-(l3)));
let g4: int = -(670);
let h4: float = -(46.27409083863361);
772;
-(962);
let i4: float = -(18.714568867529124);
431;
let j4: bool = (l3) != (-(41.26380993419628));
let k4: float = -(h4);
996;
let l4: float = k4;
24.122176411982423;
let m4: float = -(16.348836844943012);
let n4: float = -(15.948134539752955);
let o4: int = -(708);
let p4: float = -(k4);
let q4: int = (761) * (-(o4));
g4;
h4;
let r4: float = -(k4);
123;
let s4: int = -(((-(q4)) - (o4)) - (-(647)));
(-(117)) + (-(672));
let t4: int = -(356);
false;
let u4: float = 18.359735759113637;
let v4: bool = !((-(26.125819258980144)) != (i4));
!(p3);
t4;
false;
let w4: bool = p3;
let x4: bool = v4;
let a5: int = -(713);
let b5: bool = !(false);
(908) >= ((-(g4)) / (-(688)));
-(k4);
let c5: float = -(h4);
!((-(977)) < (-(416)));
let d5: bool = true;
-(200);
-(954);
!(true);
-(170);
-(o4);
let e5: int = (72) - (-(312));
-(t4);
let f5: float = -(24.97978648081179);
let g5: int = -(g4);
let h5: int = -(g5);
-(560);
let i5: int = o4;
-(37.94625289961057);
m4;
!(false);
p3;
-(2.6401763209013605);
let j5: bool = !((-(839)) <= (-((-(h5)) * (-((-(h5)) - ((793) / (u4)))))));
!(d5);
let k5: int = -(619);
a5;
let l5: int = if !(true) {
  let m5: float = -(49.49534115546946);
  s4 + o4 + i5 + t4 + g4 + h5 + a5 + q4
} else if !(x4) {
  b5;
  let n5: bool = !((224) > (213));
  let o5: int = -(966);
  let p5: int = i5;
  let q5: float = 47.67325235077761;
  -(i4);
  -((u4) / (-(398)));
  (686) + (o5);
  o4 + e5
} else if true {
  -(l4);
  let r5: bool = (j5) != (w4);
  (239) / (-(257));
  let s5: bool = !(m2);
  n4;
  p3;
  let t5: bool = m2;
  let u5: bool = !(v4);
  g5 + h5 + a5 + i5 + q4 + s4 + o4
} else {
  let v5: int = -((951) / (h5));
  u4;
  h4;
  true;
  !(true);
  -(k5);
  -((500) - (-((-(g4)) % (-(31)))));
  486;
  i5 + q4 + e5 + v5
};
t4;
let w5: float = 46.27925972553099;
278;
-(m4);
let x5: bool = !(p3);
let a6: float = -(11.72738479654873);
q4;
let b6: bool = !(v4);
-(30.361310393504414);
526;
-(g5);
!((415) == (-((382) % (782))));
-(l5);
e5;
m2;
v4;
-(g5);
(759) != (-((-(e5)) - (962)));
-(1.3194606431634563);
-(i5);
!(j5);
let c6: bool = !(true);
a5;
let d6: bool = !(true);
let e6: bool = (m4) == (37.05941176612143);
565;
-(148);
let f6: int = g4;
-(10.129823901561984);
!(if !(d6) {
  !(true);
  let g6: int = -(f6);
  let h6: float = -(r4);
  !((p4) < (-(584)));
  true;
  !(if true {
    -(g5);
    !(b6);
    false;
    false
  } else if true {
    let i6: bool = !(e6);
    !(i6);
    -(619);
    false;
    let j6: int = -(479);
    let k6: int = 3;
    !((!(false)) != (j4))
  } else if (8.559965552456548) > (24.03479518716716) {
    -(24.155091057483634);
    k4;
    48.36231906961264;
    true;
    x4;
    b5
  } else if !(m2) {
    !(e6)
  } else {
    false;
    let l6: int = g6;
    -(s4);
    let m6: int = 430;
    !(false)
  })
} else if true {
  let n6: int = -(if !(true) {
    let o6: int = 442;
    let p6: float = -(33.512664460155385);
    g5 + k5 + g4 + h5
  } else if d6 {
    let q6: bool = d6;
    -((-(s4)) - (-(164)));
    -(a6);
    k5 + t4 + i5 + e5
  } else if !(d5) {
    j5;
    t4;
    let r6: float = -(f5);
    !(j4);
    -(r4);
    p4;
    o4 + h5 + l5 + g5 + e5 + k5 + q4 + i5 + a5 + s4
  } else {
    let s6: float = l4;
    16.520803731704248;
    27.503285797210975;
    -(108);
    m4;
    -((s4) + ((g5) + (679)));
    let t6: float = f5;
    t4 + a5
  });
  i4;
  -(25);
  !(w4)
} else if true {
  !(j5)
} else if !(x5) {
  !(true);
  let u6: float = u4;
  let v6: int = -(o4);
  -(l4);
  -(45.00174781935433);
  (23.26404623007657) == (-(i4));
  let w6: int = -(617);
  false
} else {
  let x6: int = -(i5);
  let a7: float = a6;
  !(true)
});
let b7: int = -(t4);
x4;
let c7: int = -(879);
let d7: bool = !(true);
179;
if true {
  e5 + c7 + q4 + g5 + f6 + k5
} else {
  let e7: int = -(475);
  let f7: int = -(e5);
  let g7: int = g4;
  let h7: int = k5;
  let i7: float = u4;
  q4;
  w5;
  308;
  g7 + f6 + o4 + s4 + e7
};
430;
let j7: int = 137;
let k7: int = 242;
let l7: int = s4;
!(m2);
!(v4);
let m7: int = if true {
  -(k4);
  let n7: bool = j5;
  m2;
  s4 + t4 + j7 + e5 + a5 + o4 + k5 + g5 + h5 + c7 + k7 + g4 + l7 + q4
} else if false {
  557;
  let o7: float = 49.42964187938009;
  l7 + g4
} else if false {
  let p7: bool = (-(39.143358797651516)) != (-(32.156623786843056));
  let q7: bool = !(if false {
    -(h4);
    -(47.01382921700802);
    let r7: int = -((-(o4)) - (-(200)));
    let s7: float = -(p4);
    !(p7)
  } else if true {
    -(13.92692321093042);
    let t7: bool = !((-(27.96576602470598)) == (-(45.829280203496005)));
    !(w4)
  } else {
    !((808) != (i5))
  });
  g4 + h5
} else if (p4) <= (-(998)) {
  let u7: float = -(r4);
  -(h4);
  23.959023458207138;
  let v7: int = l7;
  c7 + h5 + b7 + g5 + l7 + a5 + i5 + f6 + l5 + q4 + k5 + v7 + k7 + o4 + g4 + j7 + t4 + s4 + e5
} else {
  k4;
  f6 + a5 + h5 + e5 + i5 + k7 + s4 + q4 + c7 + o4 + j7 + l5 + b7
};
-(n4);
true;
8.431214609197495;
!((l7) > (-((-(933)) - (-(f5)))));
!(true);
!(false);
let w7: float = l4;
let x7: bool = !(false);
-(w5);
let a8: bool = true;
-((-(h5)) - (-(l5)));
let b8: float = c5;
let c8: int = 495;
let d8: float = w5;
i5;
206;
false;
let e8: int = i5;
-(d8);
false;
true;
let f8: bool = (-(43.47156600261709)) == (26.728650442533745);
!(false);
let g8: int = 717;
let h8: float = i4;
4.512353993167793;
271;
!(false);
let i8: float = f5;
!(b6);
i5;
let j8: bool = !(if !(a8) {
  let k8: bool = !(false);
  d5;
  !(false)
} else if (-(s4)) < (-((912) / (-(454)))) {
  !(true);
  let l8: bool = (!((if !(p3) {
    431;
    m2;
    !((-(220)) == (562))
  } else if false {
    l5;
    let m8: bool = !(false);
    let n8: float = -(b8);
    false;
    !(false)
  } else {
    let o8: int = (-(183)) + (-(o4));
    (-(h4)) == (42.19467150689327);
    let p8: int = -(g8);
    i5;
    -(129);
    -(g5);
    d5;
    let q8: float = b8;
    true
  }) == (!(b6)))) != (!(false));
  v4;
  -(c8);
  false
} else if b5 {
  -(m7);
  let r8: int = -(j7);
  !((-(662)) < (-(k4)))
} else if (!((46.6016340070433) != (-(16.788808972448145)))) == (!((-(905)) < (h4))) {
  let s8: bool = !((-(o4)) <= (29.069245791113065));
  -(864);
  !(if !(j4) {
    s8;
    let t8: float = (595) * (i4);
    -(8.946091566244721);
    !((!(true)) == ((if !(false) {
      -(c5);
      -(k4);
      false;
      let u8: int = -(c8);
      !(j5);
      !(false);
      a5 + c8 + f6 + h5 + g8 + k5 + c7 + e5 + l5 + o4 + u8 + g5 + e8 + t4 + q4
    } else if m2 {
      let v8: float = 32.77173007587102;
      m7 + c7 + e5 + h5 + k7 + g8 + e8 + s4 + t4 + g5 + k5 + o4 + q4 + i5 + j7 + l7 + a5
    } else if !(w4) {
      let w8: float = -(45.36902553409747);
      !(m2);
      !(false);
      b6;
      -(12.14672277186532);
      let x8: bool = !(x5);
      let a9: bool = (563) < (g4);
      let b9: float = 14.039738953955776;
      f6 + b7 + g5 + l5 + c8 + q4 + g4 + o4 + a5 + s4 + g8 + i5 + k7 + l7 + e8 + k5 + t4 + m7 + e5 + c7
    } else if !(false) {
      let c9: int = j7;
      q4
    } else if !(false) {
      !(false);
      !(true);
      let d9: float = -(31.05737194677602);
      let e9: bool = !(true);
      h8;
      let f9: int = m7;
      d9;
      -(38.573961900813906);
      h5 + g8 + k5 + e8 + c8 + l7 + f9 + q4 + b7 + k7 + s4 + g4 + o4 + l5 + e5 + c7 + g5
    } else if p3 {
      487;
      k4;
      (l5) + (f6);
      let g9: float = r4;
      let h9: bool = !(false);
      !(false);
      let i9: float = 33.77306836432036;
      k5
    } else {
      -(i4);
      false;
      let j9: int = -(k7);
      let k9: float = -(l3);
      let l9: int = -((e5) + ((-(985)) % (-(173))));
      h5 + g8 + o4 + g5 + g4 + l7 + i5 + q4 + k7 + m7 + t4 + l5 + s4 + e8 + j7 + k5 + e5 + c8 + l9 + a5 + b7 + f6 + j9 + c7
    }) >= (17.109263521618253)));
    j7;
    let m9: float = 19.327490289500844;
    false;
    !((-(m7)) > (-(637)))
  } else if true {
    (false) != (w4);
    let n9: int = -(426);
    !((true) == (!(a8)));
    -(i4);
    let o9: float = h8;
    -(j7);
    !(b6)
  } else if true {
    let p9: float = -(33.37656528139459);
    let q9: int = -(e8);
    let r9: float = 31.003932921997258;
    352;
    !(true);
    d6;
    let s9: bool = false;
    -(a6);
    !(true)
  } else if false {
    let t9: float = -(n4);
    true;
    let u9: int = -(j7);
    -(u4);
    24;
    let v9: float = p4;
    l7;
    p3
  } else if c6 {
    let w9: float = -(18.293160732347754);
    129;
    if !(true) {
      let x9: float = w5;
      let a10: int = -(b7);
      let b10: int = -(l5);
      if (d8) <= (-(25.554123893900094)) {
        a10;
        !(false);
        let c10: int = c8;
        2.6208655687769533;
        !(x5)
      } else {
        let d10: float = -(l4);
        !(if x5 {
          false;
          708;
          e8;
          !(false);
          let e10: int = j7;
          -(k4);
          let f10: int = c7;
          !((-((-(279)) - (-((32.14597355497977) * (437))))) != (-(19.748487821500373)))
        } else if (-(18.41318064523026)) >= (-(f5)) {
          c7;
          let g10: bool = true;
          let h10: float = 25.188810392344195;
          p4;
          true
        } else if w4 {
          -(180);
          let i10: float = n4;
          !(j4)
        } else {
          let j10: float = i8;
          e6;
          false;
          b8;
          false;
          (x9) == (46.33623889300254)
        })
      };
      !(true);
      -(j7);
      let k10: float = -(48.22977444626827);
      let l10: int = a5;
      !(false)
    } else if (-(c7)) > ((-(142)) - (-(a5))) {
      let m10: int = (627) / (-(332));
      743;
      i5;
      !((-(b8)) != (n4));
      ((-(i5)) * (-(k7))) > (971);
      -((-(u4)) / (-(986)));
      true;
      s8
    } else {
      let n10: float = 3.167461490887422;
      !(p3);
      let o10: bool = x4;
      -(m7);
      !((-(813)) > (-(f6)));
      let p10: int = 578;
      let q10: float = -(24.350489104021943);
      -(16.176759816848666);
      v4
    }
  } else if !(true) {
    let r10: bool = !((45.01080706845364) <= (k5));
    let s10: float = k4;
    !((-(f6)) == (t4))
  } else {
    !(w4);
    false
  })
} else if a8 {
  -(c7);
  let t10: float = n4;
  p3;
  let u10: float = -(d8);
  !(x7);
  -((-(24)) * (441));
  p3
} else if !(false) {
  14.63236356674873;
  let v10: float = 36.25031877017668;
  g5;
  48;
  !(true);
  771;
  true
} else {
  let w10: bool = !(j5);
  true;
  let x10: int = 186;
  let a11: bool = !(true);
  -(46.901416418147);
  !(d5);
  -(371);
  -(d8);
  !(d5)
});
let b11: float = -(w7);
let c11: int = -(889);
u4 + h4 + a6 + c5 + r4 + w5 + b11 + i4 + n4 + d8 + w7 + m4 + b8 + i8 + f5 + l4 + k4 + b11 + w5 + i8 + m4 + r4 + h8 + p4 + n4 + d8 + u4 + i4 + f5 + h4 + l4 + a6 + l3