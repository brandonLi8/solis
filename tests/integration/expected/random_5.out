-4589198922156282843